app/controllers/products_controller.rb

```diff
@@ -0,0 +1,86 @@
+class ProductsController < ApplicationController
+
+  def index
+  @sellers = show_sellers
+  @categories =  show_category
```

Comment:
Instead of the two method calls above (show_sellers and show_category), you could have a before_action to set these two instance variables. Also, the methods themselves attempt to set these same instance variables (@sellers and @categories), which doesn't make sense because you are assigning the return values of those methods to the same instance variables here. An alternative to using a before_action is to define the show_sellers and show_category methods in the models and have them return what you want these two instance variables to be set to. 🐸